### ISSUE TITLE
Explicit call of magic method: should give warning?

### ISSUE THREAD

**Issue by pylint-bot** [PyCQA/pylint#325]
Originally reported by: **Łukasz Rogalski (BitBucket: [rogalski](http://bitbucket.org/rogalski), GitHub: @rogalski?)**

----------------------------------------

In typical cases, explicit call to magic method should be considered as violation. Consider this example:
```
#!python
class Comparable(object):
    def __cmp__(self, other):
        return some_integer

cmp1 = Comparable()
cmp2 = Comparable()

print cmp1 == cmp2 # correct
print cmp1 > cmp2 # correct
print cmp1.__cmp__(cmp2) # violation

```

----------------------------------------
- Bitbucket: https://bitbucket.org/logilab/pylint/issue/327

**Comment by PCManticore** [comment]
Closing, since it's not necessarily a problem and it's a bit pedantic.
